### PR TITLE
fix: Update Gas Services docset

### PR DESCRIPTION
**Pull request by ffe9f8**

### CODE REVIEW

File: src/pages/dev/general-message-passing/gas-services/intro.mdx [hunk 1]
```diff
@@ -1,17 +1,43 @@
...
+The Axelar network routes every [General Message Passing (GMP)](https://docs.axelar.dev/dev/general-message-passing/overview) transaction to its destination chain. The final step of this process involves the application paying the gas fees for the transaction. This can be done in one of two ways:
 
-## Gas Receiver
-For any General Message Passing (GMP) transaction, the Axelar network routes transactions to their destination chains. Execution is the final step of the pipeline to the specified destination contract address on the destination chain. It is invoked in one of two ways.
+1. **On the destination chain:** Manually paid on the destination chain
+1. **On the source chain:** Prepaid to the `AxelarGasService` contract on the source chain, thus allowing Axelar to execute the final transaction on the destination chain
```
Comment (StephenFluin) [inline src/pages/dev/general-message-passing/gas-services/intro.mdx:10]:
```suggestion
1. **On the source chain:** Prepaid to the `AxelarGasService` contract on the source chain, thus allowing a relayer to execute the final transaction on the destination chain
```

File: src/pages/dev/general-message-passing/gas-services/intro.mdx [hunk 2]
```diff
@@ -1,17 +1,43 @@
...
 
-Gas Receiver is a smart contract deployed on every EVM that is provided by Axelar. Our gas services provide users/applications the ability to prepay the full estimated cost for any GMP transaction (from source to destination chains) with the convenience of a single payment on the source chain, relying on Axelar's relay services to manage the full pipeline. Once gas is paid to Gas Receiver for a GMP transaction, Axelar's relayer services pick up the payment and automatically execute the final General Message Passing call.
+## Increase gas
 
-Developers can use Axelar gas services by prepaying upfront the relayer gas fee on the source chain, thereby covering the cost of gas to execute the final transaction on the destination chain.
\ No newline at end of file
+If the amount of prepaid gas is not enough — for example, if the destination chain is congested with transfers, causing a gas fee increase for each transaction — the application can resubmit a new amount of gas to pay for the transaction.
```
Comment (StephenFluin) [inline src/pages/dev/general-message-passing/gas-services/intro.mdx:16]:
```suggestion
If the amount of prepaid gas is not enough — for example, if the destination chain is congested with transfers, causing a gas fee increase for each transaction — the application can add additional gas to pay for the transaction.
```

File: src/pages/dev/general-message-passing/gas-services/intro.mdx [hunk 3]
```diff
@@ -1,17 +1,43 @@
...
+
+## The Executor Service
```
Comment (StephenFluin) [inline src/pages/dev/general-message-passing/gas-services/intro.mdx:18]: We should simplify and only talk about the "Relay Service" "Executor Service" or "Gas Service". I prefer gas service.

File: src/pages/dev/general-message-passing/gas-services/pay-gas.mdx [hunk 4]
```diff
@@ -50,92 +50,47 @@ contract SimpleTransferContract {
 }
 ```
 
-The `msg.value` is the gas amount we pay to the `AxelarGasService` contract.
-
-So, on the front-end side, we need to pass `sourceGasFee` to `msg.value` like below:
+`msg.value` is the gas amount paid to the `AxelarGasService` contract. Thus, `sourceGasFee` must be passed to `msg.value` on the front end:
```
Comment (StephenFluin) [inline src/pages/dev/general-message-passing/gas-services/pay-gas.mdx:53]: I want to clarify that you aren't passing a variable in, you need to pass the VALUE you got from `sourceGasFee`. Any ideas?

File: src/pages/dev/general-message-passing/gas-services/pay-gas.mdx [hunk 5]
```diff
@@ -50,92 +50,47 @@ contract SimpleTransferContract {
...
-    bytes calldata payload,
-    address refundAddress
-) external payable;
-```
-
-## Alternative gas payment methods for `callContractWithToken`
+After sending a transaction out, Axelar’s Executor Service will do the following:
 
+- Monitor `AxelarGasReceiver` for receipt of payment and get the amount paid as `amountPaid`.
+- Match the gas paid to the correct contract calls.
```
Comment (StephenFluin) [inline src/pages/dev/general-message-passing/gas-services/pay-gas.mdx:64]: The relayer also submits the transaction to the Axelar network for confirmation

File: src/pages/dev/general-message-passing/gas-services/pay-gas.mdx [hunk 6]
```diff
@@ -50,92 +50,47 @@ contract SimpleTransferContract {
...
-- For `payNativeGasForContractCallWithToken`, the following need to match the arguments of a `contractCallWithToken` on the `AxelarGateway`:
-  - `destinationChain`
-  - `destinationAddress`
-  - `payload`
-  - `symbol`
-  - `amount`
-- For `payNativeGasForContractCall` and `payNativeGasForContractCallWithToken`, the amount of funds received is specified by `msg.value`.
-- For all functions, `refundAddress` is the address that will eventually be able to receive excess amount paid for gas.
+- `sender` must match the address that calls `callContract()` or `callContractWithToken()` on the `AxelarGateway`. If the `AxelarGasReceiver` is called by the same contract that will call the Gateway, specify `address(this)` as `sender`.
+- `refundAddress` is the address that will eventually be able to [receive the excess amount paid for gas](https://docs.axelar.dev/dev/general-message-passing/gas-services/refund).
```
Comment (StephenFluin) [inline src/pages/dev/general-message-passing/gas-services/pay-gas.mdx:74]: Should we clarify that refunds are made on the source chain?

File: src/pages/dev/general-message-passing/gas-services/refund.mdx [hunk 7]
```diff
@@ -1,16 +1,9 @@
-# Refund the prepaid gas
+# Refund Prepaid Gas
 
-The prepaid gas to `payNativeGasForContractCall` or `payNativeGasForContractCallWithToken` could exceed the actual amount needed for relaying a message to the destination contract.
+If the gas fees prepaid to `payNativeGasForContractCall()` or `payNativeGasForContractCallWithToken()` exceeds the actual amount needed for relaying a message to the destination contract, the Executor Service will automatically track the excess gas amount and refund it to the payer’s wallet address by calling `refund()` on the [`GasService` contract](https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/ad37802dc6d62fff3ab589f2605f7a3e566977dd/contracts/interfaces/IAxelarGasService.sol).
```
Comment (StephenFluin) [inline src/pages/dev/general-message-passing/gas-services/refund.mdx:3]:
```suggestion
If the gas fees prepaid to `payNativeGasForContractCall()` or `payNativeGasForContractCallWithToken()` exceeds the actual amount needed for relaying a message to the destination contract, the Executor Service will automatically caclculate the excess gas amount and refund it to the payer’s wallet address by calling `refund()` on the [`GasService` contract](https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/ad37802dc6d62fff3ab589f2605f7a3e566977dd/contracts/interfaces/IAxelarGasService.sol).
```

**Review by StephenFluin — CHANGES_REQUESTED** [review]

File: src/pages/dev/general-message-passing/gas-services/intro.mdx (hunk 3 above)
Comment (ffe9f8) [reply src/pages/dev/general-message-passing/gas-services/intro.mdx:18]: That sounds good to me. I think we should default to "Gas Service" as well (proper noun). Will update the style guide accordingly.

File: src/pages/dev/general-message-passing/gas-services/pay-gas.mdx (hunk 4 above)
Comment (ffe9f8) [reply src/pages/dev/general-message-passing/gas-services/pay-gas.mdx:53]: I removed references to `sourceGasFee` entirely (except in a snippet where it's clearly marked as a variable that stores the return value of `estimateGasFee()`) and replaced it with references to call `estimateGasFee()`.

File: src/pages/dev/general-message-passing/gas-services/pay-gas.mdx (hunk 6 above)
Comment (ffe9f8) [reply src/pages/dev/general-message-passing/gas-services/pay-gas.mdx:74]: Updated here and in the intro

**Review by StephenFluin — APPROVED** [review]

**Pull request merged**